dd-trace-api/src/main/java/com/signalfx/tracing/api/TraceSetting.java

```diff
@@ -0,0 +1,20 @@
+package com.signalfx.tracing.api;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Use this annotation for class and method per-instrumentation configurations
+ *
+ */
+@Retention(RUNTIME)
+@Target({TYPE, METHOD})
+public @interface TraceSetting {
+
+  /** An array of Exceptions to ignore in jax-rs-controller auto-instrumented spans. */
```

Comment:
Comment says "jax-rs-controller" but this is in a generic package.  Think carefully about whether this is generic enough to share across multiple instrumentations.  Might be better to make this a jax-rs-controller specific annotation until more consideration is put into a generic one.  The fact that you commented it the way you did makes me think you mean for it to be jax-rs-controller specific for now.

Reply:
I have thought about this being adopted by other instrumentations, as this use particular whitelist case can be extended for on method exit throwable checks project wide.  The comment is to denote that so far only the jax-rs annotation instrumentation has adopted it, and would be updated as that changes.

The trace setting annotation overall is designed to be a generic runtime configuration helper as providing one per-instrumentation doesn't scale.